src/streamlink/logger.py

```diff
@@ -55,55 +44,34 @@ def makeRecord(self, name, level, fn, lno, msg, args, exc_info,
         A factory method which can be overridden in subclasses to create
         specialized LogRecords.
         """
-        if name.startswith("streamlink"):
-            rv = _LogRecord(name, level, fn, lno, msg, args, exc_info, func, sinfo)
-        else:
-            rv = _CompatLogRecord(name, level, fn, lno, msg, args, exc_info, func, sinfo)
+        rv = LogRecord(name, level, fn, lno, msg, args, exc_info, func, sinfo)
```

Comment:
`def makeRecord` could also be removed

https://github.com/python/cpython/blob/3.9/Lib/logging/__init__.py#L1538-L1551

maybe there is even more, that could be removed